### PR TITLE
Add explicit role to stacked layout message

### LINKED ISSUE #3537: Accessibility issue : Ensure aria-roledescription is only used on elements with an implicit or explicit role
[edit by @corinagum 11/9/2020]

See latest [comment](https://github.com/microsoft/BotFramework-WebChat/issues/3537#issuecomment-724381338): I have reopened this bug due to 'layout double' and 'layout carousel' still showing as breaking accessibility in Accessibility Insights.

/end edit

---

Running a fast pass on WebChat bot, getting this error

Path | #webchat__stacked-layout__id--i5ta9

<div aria-roledescription="message" class="webchat__stacked-layout__message-row" id="webchat__stacked-layout__id--i5ta9">
**Fix the following:Give the element a role that supports aria-roledescription**

Similarly on all these:

Path | #webchat__stacked-layout__id--wpsrv
#webchat__stacked-layout__id--y6f93
#webchat__stacked-layout__id--vsynk

image

### PR DESCRIPTION
**Pull request by corinagum**
> Fixes #3537

## Changelog Entry
-  Fixes [#3537](https://github.com/microsoft/BotFramework-WebChat/issues/3537). [Accessibility]: Ensure `aria-roledescription` is only used on elements with implicit/explicit role based off of [WAI role attributes](https://www.w3.org/WAI/PF/HTML/wiki/RoleAttribute), by [@corinagum](https://github.com/corinagum) in PR [#3551](https://github.com/microsoft/BotFramework-WebChat/pull/3551)

## Description

- Accessibility Insights for Web showed messages as having no implicit/explicit role. This change adds explicit `role="group"` to messages.
- Although WAI-ARIA indicates that `role="group"` can have a child `role="listitem"`, Accessibility Insights files the pattern as an a11y bug. 

-  [x] I have added tests and executed them locally
-  [x] I have updated `CHANGELOG.md`
-  [ ] ~I have updated documentation~

## Review Checklist

> This section is for contributors to review your work.

-  [x] Accessibility reviewed (tab order, content readability, alt text, color contrast)
-  [x] Browser and platform compatibilities reviewed
-  [x] CSS styles reviewed (minimal rules, no `z-index`)
-  [x] Documents reviewed (docs, samples, live demo)
-  [x] Internationalization reviewed (strings, unit formatting)
-  [x] `package.json` and `package-lock.json` reviewed
-  [x] Security reviewed (no data URIs, check for nonce leak)
-  [x] Tests reviewed (coverage, legitimacy)

### CODE REVIEW

File: packages/component/src/Activity/StackedLayout.js [hunk 1]
```diff
@@ -162,6 +162,7 @@ const StackedLayout = ({
               // Disable "Prop `id` is forbidden on DOM Nodes" rule because we are using the ID prop for accessibility.
               /* eslint-disable-next-line react/forbid-dom-props */
               id={ariaLabelId}
+              role="log"
```
Comment (compulim) [inline packages/component/src/Activity/StackedLayout.js:165]: First, `role="log"` implies `aria-live="polite"`. We shouldn't use this role.

Second, this is the message bubble, not a transcript.

**Review by compulim — CHANGES_REQUESTED** [review]

File: packages/component/src/Activity/StackedLayout.js (hunk 1 above)
Comment (corinagum) [reply packages/component/src/Activity/StackedLayout.js:165]: Then what do you suggest?

Comment (compulim) [reply packages/component/src/Activity/StackedLayout.js:165]: https://www.w3.org/WAI/PF/aria/roles

Let's pick one of the role here.

**Review by compulim — COMMENTED** [review]
Please add test too.

File: packages/component/src/Activity/StackedLayout.js [hunk 2]
```diff
@@ -152,7 +152,7 @@ const StackedLayout = ({
       })}
       role="group"
     >
-      <div className="webchat__stacked-layout__main">
+      <div className="webchat__stacked-layout__main" role="list">
```
Comment (compulim) [inline packages/component/src/Activity/StackedLayout.js:155]: "list" implies it's a container with multiple items to be narrated.

While it is true that activity could have multiple attachments, the text message itself is not part of attachments. Activity = text message + N x attachment.

So I think "listitem" is not a good role for text message as it is always single.

Comment (corinagum) [reply packages/component/src/Activity/StackedLayout.js:155]: Where does attachment come into this change? An activity has an ordered list of children, one of which is the message. There are various parts to the activity, but the message is the main child of that list.

**Review by cwhitten — APPROVED** [review]

**Pull request merged**